Features/MostPopular/MostPopularViewController.swift

```diff
@@ -41,7 +41,10 @@ class MostPopularViewController: UIViewController, MostPopularViewDelegate {
         
         loadMovieData()
         
+//        The app crashed if there were no movies(Error: Failed to load from server), so I added this check to prevent the crash.
         DispatchQueue.main.async {
+            guard !self.movies.isEmpty else { return }
```

Comment:
This will solve the issue, but I want to understand exactly what is causing the crash. Can you dig a bit to narrow where the crash is happening? If this is not easily reproducible you can add a small delay to the API response: go to `MovieDataService`, at line 75, replace it with this:
```swift
DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
    completion(.success(moviesData.results))
}
```
It will delay the success response by 1 second.

Reply:
It crashed because the vc has to scroll to a certain item in the array. 
`var itemInViewIndex: Int = 0`
`self.contentView.collectionView.scrollToItem(at: IndexPath(item: self.itemInViewIndex, section: 0), at: UICollectionView.ScrollPosition.centeredHorizontally, animated: false)`

 Since there are no items, it crashed.